ibis/pandas/execution/tests/test_window.py

```diff
@@ -582,3 +584,124 @@ def count_both(v1, v2):
+                group_by=kwds.get('group_by', self._group_by),
+                order_by=kwds.get('order_by', self._order_by),
+                preceding=kwds.get('preceding', self.preceding),
+                following=kwds.get('following', self.following),
+                max_lookback=kwds.get('max_lookback', self.max_lookback),
+                how=kwds.get('how', self.how),
+            )
+            return CustomWindow(**new_kwds)
+
+    class CustomAggContext(AggregationContext):
+        def __init__(
+            self, parent, group_by, order_by, dtype, max_lookback, preceding
+        ):
+            super().__init__(
+                parent=parent,
+                group_by=group_by,
+                order_by=order_by,
+                dtype=dtype,
+                max_lookback=max_lookback,
+            )
+            self.preceding = preceding
+
+        def agg(self, grouped_data, function, *args, **kwargs):
+            upper_indices = pd.Series(range(len(self.parent) + 1)) + 1
+            window_sizes = (
+                grouped_data.rolling(self.preceding.value + 1)
+                .count()
+                .reset_index(drop=True)
+            )
+            lower_indices = upper_indices - window_sizes
+            mask = ~upper_indices.isna()
+
+            result = window_agg_udf(
+                grouped_data,
+                function,
+                lower_indices,
+                upper_indices,
+                mask,
+                self.dtype,
+                self.max_lookback,
+                *args,
+                **kwargs,
+            )
+
+            return result
+
+    custom_window_0 = CustomWindow(
+        preceding=CustomInterval(0),
+        following=0,
+        group_by='dup_ints',
+        order_by='plain_int64',
+    )
+
+    custom_window_1 = CustomWindow(
+        preceding=CustomInterval(1),
+        following=0,
+        group_by='dup_ints',
+        order_by='plain_int64',
+    )
+
+    # Unfortunately we cannot unregister these because singledispatch
+    # doesn't support it, but this won't cause any issues either.
+    @get_preceding_value.register(CustomInterval)
+    def get_preceding_value_custom(preceding):
+        return preceding
+
+    @get_aggcontext.register(CustomWindow)
+    def get_aggcontext_custom(
+        window, *, operand, operand_dtype, parent, group_by, order_by
+    ):
+        return CustomAggContext(
+            parent=parent,
+            group_by=group_by,
+            order_by=order_by,
+            dtype=operand_dtype,
+            max_lookback=window.max_lookback,
+            preceding=window.preceding,
+        )
+
+    result_0 = my_sum(t['plain_float64']).over(custom_window_0).execute()
+    result_1 = my_sum(t['plain_float64']).over(custom_window_1).execute()
+
+    expected_0 = pd.Series([4.0, 6.0, 5.0])
+    expected_1 = pd.Series([4.0, 10.0, 5.0])
+
+    tm.assert_series_equal(result_0, expected_0)
+    tm.assert_series_equal(result_1, expected_1)
```

Comment:
Not sure if it'd be more conventional to use a fixture for all the custom objects, and parameterize this. But happy to have it like this, if we don't expect to reuse all the custom objects.

Reply:
can also parametrize this test with the different cases

Reply:
Made those fixtures